tests/test_tracer.py

```diff
@@ -461,3 +463,17 @@ def test_only_root_span_runtime(self):
         self.assertEqual(root.get_tag('language'), 'python')
 
         self.assertIsNone(child.get_tag('language'))
+
+    def test_thread_to_span(self):
+        root = self.start_span('root')
+        assert self.tracer.thread_to_span[compat.thread_get_ident()] == root
+
+        def start_span(t):
+            t.span2 = t.start_span('thread2')
+        th = threading.Thread(target=start_span, args=(self,))
+        th.start()
+        th.join()
+
+        span2 = self.span2
+        del self.span2
+        assert self.tracer.thread_to_span[th.ident] == span2
```

Comment:
If we call `span2.finish()`, `th.ident` should be removed from `self.tracer.thread_to_span` correct?

Reply:
Not necessarily; it'll be removed as soon there's no ref to `span2` and once the GC has run.